pkg/agent/controller/agent.go

```diff
@@ -219,6 +287,54 @@ func (a *Controller) serviceToRemoteServiceImport(obj runtime.Object, op syncer.
 	return serviceImport, false
 }
 
+func (a *Controller) endpointToRemoteServiceImport(obj runtime.Object, op syncer.Operation) (runtime.Object, bool) {
+	ep := obj.(*corev1.Endpoints)
+	svcExport, err := a.lighthouseClient.LighthouseV2alpha1().ServiceExports(ep.Namespace).Get(ep.Name, metav1.GetOptions{})
+	if errors.IsNotFound(err) {
+		// Service Export not created yet
+		return nil, false
+	} else if err != nil {
+		// some other error. Log and requeue
+		klog.Errorf("Unable to get ServiceExport for %#v: %v", ep, err)
+		return nil, true
+	}
+
+	serviceImport, err := a.getServiceImport(svcExport)
+
+	if err != nil {
+		// Requeue
+		return nil, true
+	}
+
+	if serviceImport.Spec.Type == lighthousev2a1.Headless && a.globalnetEnabled {
+		klog.Errorf("Headless Services not supported with globalnet yet")
+		return nil, false
+	}
+
+	oldStatus := serviceImport.Status.DeepCopy()
+	ipList := getIpsFromEndpoint(ep)
+
+	if serviceImport.Spec.Type == lighthousev2a1.SuperclusterIP && len(ipList) > 0 {
+		// Once we cache service and related objecs, this won't be needed
+		ipList = []string{serviceImport.Annotations[clusterIP]}
+	}
```

Comment:
I'm not clear on this. Shouldn't this code only run for Headless since for SuperclusterIP we've already set the ServiceImport IP(s) to ether the service's ClusterIP (or GN IP) in `getIpsForService`? I'd think we'd do:

```
if serviceImport.Spec.Type == lighthousev2a1.SuperclusterIP {
    return nil, false
}
```

after L307.

Reply:
This is to handle following use case:

1. There are no healthy pods, remove `ClusterIP` from `ServiceImport`
2. At least one pod is now healthy, add `ClusterIP` back to `ServiceImport`.

Since we remove `ClusterIP` when l`en(ipList) == 0`, we have to add it back when it becomes non-zero.

Reply:
OK - I think I get it now. Suggested comment to clarify:  "_When there's no healthy pods, the IPs in the Endpoint will become empty and thus we also clear the ServiceImport IPs to avoid clients sending a request to a cluster which has no active pods to handle the request. Once at least one backing Endpoint pod becomes healthy, we need to re-establish the ServiceImport IPs from the previously cached clusterIP annotation._"

Reply:
Done